receiver/postgresqlreceiver/config.go

```diff
@@ -0,0 +1,98 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package postgresqlreceiver
+
+import (
+	"errors"
+	"fmt"
+
+	"go.opentelemetry.io/collector/receiver/scraperhelper"
+	"go.uber.org/multierr"
+)
+
+type Config struct {
+	scraperhelper.ScraperControllerSettings `mapstructure:",squash"`
+	Username                                string   `mapstructure:"username"`
+	Password                                string   `mapstructure:"password"`
+	Databases                               []string `mapstructure:"databases"`
+	Host                                    string   `mapstructure:"host"`
+	Port                                    int      `mapstructure:"port"`
```

Comment:
Typically in the OTel collector, these are taken in together as `endpoint`. When possible, we should use `confignet.NetAddr` to provide the [endpoint](https://github.com/open-telemetry/opentelemetry-collector/blob/5007cef41b8546e5e1c1faaafd991ac5c1fe38ac/config/confignet/confignet.go#L28).

Reply:
Internally to the lib/pq connection library, `host` and `port` are separate, and are combined differently depending on if `host` begins with a `/`, implying a unix socket connection. See the library source [here](https://github.com/lib/pq/blob/v1.10.3/conn.go#L393). It didn't feel straightforward to provide them together as `endpoint`, as I don't believe the valid config of a unix socket connection would be valid as a `config.NetAddr`.

Reply:
We should have a strong bias towards using the built in config structs, unless it really doesn't make sense. We don't necessarily want the config to reflect the library that the receiver uses. Could we do something like:

```go
type Config struct {
	scraperhelper.ScraperControllerSettings `mapstructure:",squash"`
	...
	config.NetAddr `mapstructure:",squash"` // provides Endpoint and Transport
}

func (cfg *Config) Validate() error {
	var errs []error
	...

	switch cfg.Transport {
	case "tcp":
		// validation of tcp endpoint
	case "unix":
		// validation of unix endpoint
	default:
		// error
	}
	...
}
```